code/LINDA/LINDA_turf_tile.dm

```diff
@@ -358,32 +366,23 @@
 
 /datum/excited_group/proc/self_breakdown()
 	var/datum/gas_mixture/A = new
-	var/datum/gas/sleeping_agent/S = new
-	A.trace_gases += S
 	for(var/turf/simulated/T in turf_list)
-		A.oxygen 		+= T.air.oxygen
-		A.carbon_dioxide+= T.air.carbon_dioxide
-		A.nitrogen 		+= T.air.nitrogen
-		A.toxins 		+= T.air.toxins
-
-		if(T.air.trace_gases.len)
-			for(var/datum/gas/N in T.air.trace_gases)
-				S.moles += N.moles
+		A.oxygen 			+= T.air.oxygen
+		A.carbon_dioxide	+= T.air.carbon_dioxide
+		A.nitrogen 			+= T.air.nitrogen
+		A.toxins 			+= T.air.toxins
+		A.sleeping_agent 	+= T.air.sleeping_agent
+		A.agent_b 			+= T.air.agent_b
+		A.volatile_fuel 	+= T.air.volatile_fuel
 
 	for(var/turf/simulated/T in turf_list)
-		T.air.oxygen		= A.oxygen/turf_list.len
-		T.air.carbon_dioxide= A.carbon_dioxide/turf_list.len
-		T.air.nitrogen		= A.nitrogen/turf_list.len
-		T.air.toxins		= A.toxins/turf_list.len
-
-		if(S.moles > 0)
-			if(T.air.trace_gases.len)
-				for(var/datum/gas/G in T.air.trace_gases)
-					G.moles = S.moles/turf_list.len
-			else
-				var/datum/gas/sleeping_agent/G = new
-				G.moles = S.moles/turf_list.len
-				T.air.trace_gases += G
+		T.air.oxygen			= A.oxygen / turf_list.len
+		T.air.carbon_dioxide	= A.carbon_dioxide / turf_list.len
+		T.air.nitrogen			= A.nitrogen / turf_list.len
+		T.air.toxins			= A.toxins / turf_list.len
+		T.air.sleeping_agent	= A.sleeping_agent / turf_list.len
+		T.air.agent_b			= A.agent_b / turf_list.len
+		T.air.volatile_fuel		= A.volatile_fuel / turf_list.len
```

Comment:
I could probably just cache `length(turf_list)` and re-use it for faster nonsense.